jre/java/java/util/stream/Collectors.java

```diff
@@ -357,6 +364,29 @@ private static <T> Supplier<Map<Boolean, List<T>>> partitionSupplier() {
     return toMap(keyMapper, valueMapper, mergeFunction, HashMap::new);
   }
 
+  public static <T, K, U> Collector<T, ?, Map<K, U>> toUnmodifiableMap(
+      Function<? super T, ? extends K> keyMapper,
+      Function<? super T, ? extends U> valueMapper) {
+    return collectingAndThen(
+            toMap(disallowNulls(keyMapper), disallowNulls(valueMapper)),
+            Collections::unmodifiableMap
+    );
+  }
+
+  public static <T, K, U> Collector<T, ?, Map<K, U>> toUnmodifiableMap(
+      Function<? super T, ? extends K> keyMapper,
+      Function<? super T, ? extends U> valueMapper,
+      BinaryOperator<U> mergeFunction) {
+    return collectingAndThen(
+            toMap(disallowNulls(keyMapper), disallowNulls(valueMapper), mergeFunction),
+            Collections::unmodifiableMap
+    );
+  }
+
+  private static <T, R> Function<T, R> disallowNulls(Function<T, R> func) {
+    return func.andThen(Objects::requireNonNull);
```

Comment:
Sorry I wasn't clear enough earlier. I meant something like:

```
  private static <T, R> Function<T, R> disallowNulls(Function<T, R> func) {
    return x -> Object.requireNonNull(func.apply(x));
  }
```

(otherwise there would be another lambda indirection in each item during collect)

Reply:
@treblereel  I think you missed this one

Reply:
@gkdn thank you, let me take a look ...

Reply:
@gkdn fixed